### DISCUSSION TITLE
Você documenta o uso de componentes de UI?

### DISCUSSION THREAD

**Discussion by iamgbayer** [frontendbr/forum#695] [Discussão]
Olá pessoal,

Gostaria de trazer esse questionamento se seria interessante uma documentação e também caso você faça isso, de que maneira vem fazendo.

No meu contexto, hoje estamos tentando criar simples componentes atráves do Twig e Stylus, porém não documentamos o que é necessário fazer para criar um simples botão, utilizando tal mixins e outros, ficando esse entendimento **com o desenvolvedor** ou **explorando o código**.

Particularmente acho que isso poderia ser expressado em uma documentação externa ao código.

**Comment by kvnol** [comment]
Documentação é sempre importante.

O Gaiden da @getninjas é um ótimo exemplo: https://getninjas.github.io/gaiden/ :heart:

**Comment by didiraja** [comment]
Definitivamente, deve ser feito. Aqui eu sou o único desenvolvedor e mesmo assim, salvo e organizo os templates e snippets para serem usados depois.

**Comment by rsmelo** [comment]
quem já documenta, poderia apresentar algum exemplo de como costuma documentar ou algum link sobre o assunto etc

**Comment by fdaciuk** [comment]
Uma ferramenta bem legal pra fazer isso é o storybook, onde você pode criar a documentação dos seus componentes baseado nos possíveis estados que ele renderiza :)

**Comment by doug2k1** [comment]
Usamos aqui o https://react-styleguidist.js.org/ (no caso, componentes React). É bem legal. Ele lê o próprio código e comentários para gerar documentação (tipo JSDoc) e exemplos são escritos em Markdown.

**Comment by evansantos** [comment]
@doug2k1 não conhecia esse styleguidist não, achei maneiro

**Comment by ghost** [comment]
O que venho usando realmente é o storybook que o @fdaciuk mencionou. No caso como trabalho com react ele entra no workflow como uma implementação assim como o próprio componente.

Então é certo que vamos fazer:

- componente
- storybook
- testes

O lado bom dele é que não vira uma documentação obsoleta. Afinal o problema de documentações é exatamente isso e se tratando de interface mais ainda.

**Comment by MarcoWorms** [comment]
Se for react eu nem sei dizer outra ferramenta que não seja o storybook. Já utilizei em projetos e usamos aqui na pagar.me pro nossa nova dashboard em react https://github.com/pagarme/pilot funciona super bem

Eu ainda não testei, mas acho que a maneira ideal de usá-lo em um projeto grande é criar um `story.js` na mesma pasta do componente com o exemplo dele pra ir pro storybook, pq colocar tudo dentro da uma pasta chamada `storybook` fica meio ruim de mexer quando vc tem muitos componentes
